### PR TITLE
fix: wait_for_all_recoveries

### PR DESCRIPTION
**Pull request by maan2003**

### CODE REVIEW

File: fedimint-client/src/lib.rs [hunk 1]
```diff
@@ -1491,7 +1491,7 @@ impl Client {
             .wait_for(|in_progress| {
                 !in_progress
```
Comment (maan2003) [inline fedimint-client/src/lib.rs:1492]: this is hard, my head hurts
```suggestion
                in_progress
```

**Review by dpc — APPROVED** [review]

**Comment by dpc** [comment]
There's only so many combinations, we'll get it right eventually.

**Review by bradleystachurski — APPROVED** [review]

File: fedimint-client/src/lib.rs [hunk 2]
```diff
@@ -1489,9 +1489,9 @@ impl Client {
         let mut recovery_receiver = self.client_recovery_progress_receiver.clone();
         recovery_receiver
             .wait_for(|in_progress| {
-                !in_progress
+                in_progress
                     .iter()
-                    .any(|(_id, progress)| !progress.is_done())
+                    .all(|(_id, progress)| progress.is_done())
```
Comment (elsirion) [inline fedimint-client/src/lib.rs:1494]: This only improves readability, there is no functional change here.

**Review by elsirion — APPROVED** [review]

File: fedimint-client/src/lib.rs (hunk 2 above)
Comment (maan2003) [reply fedimint-client/src/lib.rs:1494]: >There's only so many combinations, we'll get it right eventually.

lol we are repeating combinations now

Comment (dpc) [reply fedimint-client/src/lib.rs:1494]: Me and Manmeet playing with `!`, `any` and `all`.

image

Comment (justinmoon) [reply fedimint-client/src/lib.rs:1494]: lmaoooo

**Pull request merged**